pczt/src/roles/creator/mod.rs

```diff
@@ -0,0 +1,164 @@
+        self.fallback_lock_time = Some(fallback);
+        self
+    }
+
+    #[cfg(feature = "orchard")]
+    pub fn with_orchard_flags(mut self, orchard_flags: orchard::bundle::Flags) -> Self {
+        self.orchard_flags = orchard_flags.to_byte();
+        self
+    }
+
+    pub fn build(self) -> Pczt {
+        Pczt {
+            global: crate::common::Global {
+                tx_version: self.tx_version,
+                version_group_id: self.version_group_id,
+                consensus_branch_id: self.consensus_branch_id,
+                fallback_lock_time: self.fallback_lock_time,
+                expiry_height: self.expiry_height,
+                coin_type: self.coin_type,
+                // Spends and outputs modifiable, no SIGHASH_SINGLE.
+                tx_modifiable: FLAG_INPUTS_MODIFIABLE | FLAG_OUTPUTS_MODIFIABLE,
+                proprietary: BTreeMap::new(),
+            },
+            transparent: crate::transparent::Bundle {
+                inputs: vec![],
+                outputs: vec![],
+            },
+            sapling: crate::sapling::Bundle {
+                spends: vec![],
+                outputs: vec![],
+                value_sum: 0,
+                anchor: self.sapling_anchor,
+                bsk: None,
+            },
+            orchard: crate::orchard::Bundle {
+                actions: vec![],
+                flags: self.orchard_flags,
+                value_sum: (0, true),
+                anchor: self.orchard_anchor,
+                zkproof: None,
+                bsk: None,
+            },
+        }
+    }
+
+    /// Builds a PCZT from the output of a [`Builder`].
+    ///
+    /// Returns `None` if the `TxVersion` is incompatible with PCZTs.
+    ///
+    /// [`Builder`]: zcash_primitives::transaction::builder::Builder
+    #[cfg(feature = "zcp-builder")]
+    pub fn build_from_parts<P: zcash_protocol::consensus::Parameters>(
+        parts: zcash_primitives::transaction::builder::PcztParts<P>,
+    ) -> Option<Pczt> {
+        use zcash_primitives::transaction::sighash::SighashType;
+        use zcash_protocol::consensus::NetworkConstants;
+
+        use crate::{common::FLAG_HAS_SIGHASH_SINGLE, SAPLING_TX_VERSION};
+
+        let tx_version = match parts.version {
+            zcash_primitives::transaction::TxVersion::Sprout(_)
+            | zcash_primitives::transaction::TxVersion::Overwinter => None,
+            zcash_primitives::transaction::TxVersion::Sapling => Some(SAPLING_TX_VERSION),
+            zcash_primitives::transaction::TxVersion::Zip225 => Some(V5_TX_VERSION),
+        }?;
+
+        // Spends and outputs not modifiable.
+        let mut tx_modifiable = 0b0000_0000;
+        // Check if any input is using `SIGHASH_SINGLE`.
+        if parts.transparent.as_ref().map_or(false, |bundle| {
+            bundle
+                .inputs()
+                .iter()
+                .any(|input| input.sighash_type() == &SighashType::SINGLE)
```

Comment:
```suggestion
                .any(|input| input.sighash_type() & !SIGHASH_ANYONECANPAY == &SighashType::SINGLE)
```
(will need rustfmt I think).